nodejs/eks/authenticationMode.test.ts

```diff
@@ -70,10 +70,20 @@ describe("validateAuthenticationMode", () => {
         };
 
         expect(() => validateAuthenticationMode(args)).toThrowError(
-            "The 'instanceRoles' property is not supported when 'authenticationMode' is set to 'API'.",
+            "The 'instanceRoles' property does not support non-empty values when 'authenticationMode' is set to 'API'.",
         );
     });
 
+    it("should not throw an error for instanceRoles=[] when authentication mode is set to API", () => {
+        const args = {
+            authenticationMode: "API",
+            instanceRoles: [],
```

Comment:
Could we also test passing instanceRoles as a Promise and Output?

Reply:
That'd be great but I was running into limitations that seem to make failing outputs un-testable. I can ask around for help.

Reply:
Might be worth adding an integration test tif that doesn't work

Reply:
I filed https://github.com/pulumi/pulumi/issues/16665 - there's something odd, possibly a bug in Node SDK that makes it extra difficult. I'll look into an integration test.